petastorm/spark/spark_dataset_converter.py

```diff
@@ -483,6 +480,47 @@ def _materialize_df(df, parent_cache_dir_url, parquet_row_group_size_bytes,
     return save_to_dir_url
 
 
+def _wait_for_fs_eventually_consistency(url_list):
+    """
+    This is used in databricks runtime. Waiting about 30 seconds to make sure
+    all files can be accessed by dbfs fuse.
+    This is because of s3 eventually consistency, and the backend of dbfs is s3.
+    """
+    fs, path_list = get_filesystem_and_path_or_paths(url_list)
+
+    wait_seconds = _get_spark_session().conf \
+        .get(SparkDatasetConverter.WAIT_FS_EVENTUALLY_CONSISTENCY_SECONDS, '0')
+    wait_seconds = int(wait_seconds)
+
+    if wait_seconds <= 0:
+        return
+
+    def wait_on_file(path):
+        beg_time = time.time()
+        while True:
+            if fs.exists(path):
+                return True
+            time.sleep(0.1)
+            elapsed = time.time() - beg_time
+            if elapsed > wait_seconds:
+                return False
+
+    pool = ThreadPool(64)
+    try:
+        async_results = [pool.apply_async(wait_on_file, (path,)) for path in path_list]
+        failed_file_list = []
+        for i in range(len(path_list)):
+            res = async_results[i].get()
+            if not res:
+                failed_file_list.append(path_list[i])
+        if failed_file_list:
+            raise RuntimeError('These files cannot be synced after waiting {wait} seconds: {path_list}'
```

Comment:
Perhaps some more context for a user: something like: `"Timeout while waiting for all parquet-store files to appear at {url}. This files cannot ...."`